java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/cp/CPMapTest.java

```diff
@@ -69,62 +96,116 @@ private String[] createCpGroupNames() {
+        state.operationCounter.setCount++;
+    }
+
+    @TimeStep(prob = 0)
+    public void put(ThreadState state) {
+        state.randomMap().put(state.randomKey(), state.randomValue());
+        state.operationCounter.putCount++;
     }
 
     @TimeStep(prob = 0)
-    public String put(ThreadState state) {
-        return state.randomMap().put(state.randomKey(), v);
+    public void putIfAbsent(ThreadState state) {
+        state.randomMap().putIfAbsent(state.randomKey(), state.randomValue());
+        state.operationCounter.putIfAbsentCount++;
     }
 
     @TimeStep(prob = 0)
-    public String get(ThreadState state) {
-        return state.randomMap().get(state.randomKey());
+    public void get(ThreadState state) {
+        state.randomMap().get(state.randomKey());
+        state.operationCounter.getCount++;
     }
 
     // 'remove' and 'delete' other than their first invocation pointless -- we're just timing the logic that underpins the
     // retrieval of no value.
 
     @TimeStep(prob = 0)
-    public String remove(ThreadState state) {
-        return state.randomMap().remove(state.randomKey());
+    public void remove(ThreadState state) {
+        state.randomMap().remove(state.randomKey());
+        state.operationCounter.removeCount++;
     }
 
     @TimeStep(prob = 0)
     public void delete(ThreadState state) {
         state.randomMap().delete(state.randomKey());
+        state.operationCounter.deleteCount++;
     }
 
     @TimeStep(prob = 0)
-    public boolean cas(ThreadState state) {
-        // 'v' is always associated with 'k'
-        return state.randomMap().compareAndSet(state.randomKey(), v, v);
+    public void cas(ThreadState state) {
+        CPMap<Integer, String> randomMap = state.randomMap();
+        Integer key = state.randomKey();
+        String expectedValue = randomMap.get(key);
+        if (expectedValue != null) {
+            randomMap.compareAndSet(key, expectedValue, state.randomValue());
+            state.operationCounter.casCount++;
+        }
     }
 
     @TimeStep(prob = 0)
     public void setThenDelete(ThreadState state) {
         CPMap<Integer, String> map = state.randomMap();
         int key = state.randomKey();
-        map.set(key, v);
+        map.set(key, state.randomValue());
         map.delete(key);
     }
 
+    @AfterRun
+    public void afterRun(ThreadState state) {
+        operationCounterList.add(state.operationCounter);
+    }
+
+    @Verify(global = true)
+    public void verify() {
+        // print stats
+        CpMapOperationCounter total = new CpMapOperationCounter();
+        for (CpMapOperationCounter operationCounter : operationCounterList) {
+            total.add(operationCounter);
+        }
+        logger.info(name + ": " + total + " from " + operationCounterList.size() + " worker threads");
+
+        // basic verification
+        for (CPMap<Integer, String> mapReference : mapReferences) {
+            int existedKeys = 0;
```

Comment:
I would say `keyCount` is number of possible keys. But `existedKeys` could be probably changed to `entryCount` (basically this is here because CPMap doesn't have anything like "size" method)